subsys/testsuite/ztest/include/ztest_test_new.h

```diff
@@ -210,6 +210,16 @@ static inline void unit_test_noop(void)
 #define Z_ZTEST(suite, fn, t_options) Z_TEST(suite, fn, t_options, 0)
 #define Z_ZTEST_F(suite, fn, t_options) Z_TEST(suite, fn, t_options, 1)
 
+/**
+ * @brief Skips the test if config is enabled
+ *
+ * Use this macro to in at the start of your test case, to skip it when
+ * config is enabledd.  Useful when your test is still under development.
+ *
+ * @param config The KConfig option used to skip the test.
+ */
+#define Z_TEST_SKIP_IFDEF(config) if (IS_ENABLED(config)) { ztest_test_skip(); }
```

Comment:
Replace the `if` statement with `COND_CODE_1(config, (ztest_test_skip()), ())` it'll be better optimized.